projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala

```diff
@@ -4,128 +4,186 @@
+      diffs
+        .foreach {
+          case (workerName, requestedStatus) =>
+            log.debug(s"Remotely requested worker [$workerName] as [$requestedStatus].")
+            actorIndex.get(workerName).foreach { workerRef =>
+              log.debug(
+                s"Setting requested status [$requestedStatus] on worker $workerName [${workerRef.path.toString}]"
+              )
+              workerRef ! requestedStatus
+            }
+        }
+
+      requestedStatusLocalCopy = remotelyChanged
+    }
+
+    // TODO: handle UpdateSuccess/UpdateFailure !!
+    case changed @ Changed(ObservedStatusDataKey) =>
+      observedStatusLocalCopy = changed.get(ObservedStatusDataKey).entries
 
-    case g @ GetSuccess(DataKey, req) =>
-      val registry: PNCounterMap[WorkerMetadata] = g.get(DataKey)
-      val desiredStatus: DesiredState            = mapStatus(registry.entries)
-      req.get.asInstanceOf[ActorRef] ! desiredStatus
+    // TODO: handle UpdateSuccess/UpdateFailure !!
+    case changed @ Changed(NameIndexDataKey) =>
+      nameIndexLocalCopy = changed.get(NameIndexDataKey).entries
 
     case Terminated(deadActor) =>
-      // update indices and stop watching
-      actorIndex = actorIndex - actorReverseIndex(deadActor)
-      actorReverseIndex = actorReverseIndex - deadActor
+      log.debug(s"Worker ${deadActor.path.name} died. Marking it as Stopped.")
+      // when a watched actor dies, we mark it as stopped. It will eventually
+      // respawn (thanks to EnsureActive) and come back to it's requested status.
+      reversedActorIndex.get(deadActor).foreach { name =>
+        updateLWWMapForObserved(name, Stopped)
+      }
+      // ... and then update indices and stop watching
+      actorIndex = actorIndex - reversedActorIndex(deadActor)
+      reversedActorIndex = reversedActorIndex - deadActor
       context.unwatch(deadActor)
 
-    // TODO: accept state changes and propagate those state changes.
-    // https://github.com/lagom/lagom/issues/1744
   }
 
-  private def mapStatus(replicatedData: Map[WorkerMetadata, BigInt]): DesiredState = {
-
-    val groupedByProjectionName: Map[String, Seq[(String, (String, BigInt))]] =
-      replicatedData.toSeq.map { case (pm, bi) => (pm.projectionName, (pm.workerName, bi)) }.groupBy(_._1)
-    val projections: Seq[Projection] =
-      groupedByProjectionName
-        .mapValues { workers: Seq[(String, (String, BigInt))] =>
-          val statusPerWorker: Seq[(String, BigInt)] = workers.toMap.values.toMap.toSeq
-          statusPerWorker
-          // TODO: below should convert a BigInt into a valid ProjectionStatus (instead of hardcoding `Running`)
-          // https://github.com/lagom/lagom/issues/1744
-            .map { case (name, bi) => ProjectionWorker(name, Started) }
-        }
-        .toSeq
-        .map { Projection.tupled }
+  private def updateLWWMapForRequests(workerMetadata: WorkerKey, requested: Status): Unit = {
```

Comment:
maybe not needed to name the chosen `ReplicatedData` in the name of the method. The fact that it's an LWWMAp is less important. 

We could have:
* updateRequestedState
* updateObservedState
* updateNameIndex

Reply:
I think I wanted to make it clear the operation involves `ddata` but I agree that the exact datastructure is not required in the name.

Reply:
In the future, we can even have a trait/implementation to isolate this and make it easier to use another data structure if necessary. But it is out of the scope for this PR though.

Reply:
> In the future, we can even have a trait/implementation to isolate this and make it easier to use another data structure if necessary. But it is out of the scope for this PR though.

That's a good idea. Maybe instead of a trait, a class encapsulating the pair of dtatastructures (the `ddata` one and the `localCopy` one).

There's quite a bit of duplication (or should I say _triplication_) in this implementation and this triplication only differs on the types of 3 `LWWMap` which we also keep a local copy of.